apimodels/cedar_models.go

```diff
@@ -1,9 +1,155 @@
 package apimodels
 
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/log"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/timber"
+	"github.com/evergreen-ci/timber/buildlogger"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/pkg/errors"
+)
+
 type CedarConfig struct {
 	BaseURL  string `json:"base_url"`
 	RPCPort  string `json:"rpc_port"`
 	Username string `json:"username"`
 	APIKey   string `json:"api_key,omitempty"`
 	Insecure bool   `json:"insecure"`
 }
+
+// GetBuildloggerLogsOptions represents the arguments passed into the
+// GetBuildloggerLogs function.
+// TODO (EVG-20019): Remove the "V2" suffix once log fetching is consolidated
+// and the original GetBuildloggerLogsOptions struct is deleted.
+type GetBuildloggerLogsOptionsV2 struct {
+	BaseURL   string   `json:"_"`
+	TaskID    string   `json:"-"`
+	Execution *int     `json:"-"`
+	TestName  string   `json:"-"`
+	Tags      []string `json:"-"`
+	Start     int64    `json:"-"`
+	End       int64    `json:"-"`
```

Comment:
Is it possible to have this instead pass `time.Time`s rather than int64s representing Unix nanoseconds? It already gets converted into a `time.Time` anyways.

If not, can we document that these are nanoseconds since the Unix epoch?

Reply:
This only gets used by `taskoutput` logs where the start and end times are `int64`s representing Unix nanoseconds, [for example](https://github.com/evergreen-ci/evergreen/blob/main/taskoutput/task_log.go#L46-L51). I can add documentation, but the idea is that no other part of the code should use this (it will also go away in a year or so after we cut over and logs are TTL'ed).

Reply:
If these are going away in a year, then this SGTM.

Since it's a temporary piece of logic, can we have a ticket to track deletion of this code in a year?

Reply:
Created https://jira.mongodb.org/browse/EVG-21010 and added a TODO.